students/Z_shen/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,143 @@
+            donor[i]['Donation'] = float(donor[i]['Donation']) + float(thx_amount)
+            thx_amount = donor[i]['Donation']
+    return thx_amount
+
+
+def thank_you_letter(donor):
+    while True:
+        user_input = str(input('''What is the name of donor? or type 'list' to see the current list: ''')).title()
+        if user_input.lower() == 'back':
+            return False
+        elif user_input.lower() == 'list':
+            donor_name_list(donor)
+            continue
+        thx_amount = input('How much did ' + user_input + ' donate? $ ')
+        if str(thx_amount.lower()) == 'back':
+            return False
+        elif user_input not in sort_donor_total_given(donor):
+            new_donor = input('Do you want to add {} as a new donor? (YES or NO) '.format(user_input.title()))
+            if new_donor.lower() == 'back':
+                return False
+            elif new_donor.upper() == 'NO':
+                continue
+            else:
+                thx_name = user_input
+                add_new_donor(thx_name, float(thx_amount), donor)
+                break
+        else:
+            thx_name = user_input
+            add_amount_same_name(thx_name, float(thx_amount), donor)
+            break
+    thx = {'Name': thx_name, 'Donation': float(thx_amount)}
+    print('\nEMAIL CONTENT: \n')
+    print('''Dear {Name}, 
+    Thank you for your generous donation of ${Donation:,.2f} to us.
+    It will be put to very good use.
+        
+                        Sincerely,
+                            -The Team
+                                      '''.format(**thx))
+
+
+def print_menu_task():
+    print('''
+    Menu of Options
+    1) Send a Thank You letter to a single donor
+    2) Create a Report
+    3）Send letters to all donors
+    4) Quit
+    Note: You can type 'back' to return to Main Menu
+    ''')
+    print()
+
+
+def input_menu_choice():
+    choice = int(input('Which choice would you to perform? [1 to 4]: '))
+    print()
+    return choice
+
+
+def exit_program():
+    print('Exit the Program')
+
+
+def send_letters_to_all(donor):
+    for name in donor.keys():
+        first, last = name.split()
+        filename = first + '_' + last
+        filename = '{}.txt'.format(filename)
+        with open(filename, 'w') as file:
+            file.write(f'''Dear {name},
+        Thank you for your generous donation of ${sum(donor.get(name)):,.2f} to us.
+        It will be put to very good use.
+
+                            Sincerely,
+                                -The Team
+                                      ''')
+    print('Thank you letter(s) sent')
+
+
+while __name__ == '__main__':
+    print_menu_task()
+    number = input_menu_choice()
+    if number == 1:
+        thank_you_letter(donor)
+    elif number == 2:
+        print_donor_report(donor)
+    elif number == 3:
+        send_letters_to_all(donor)
+    elif number == 4:
+        exit_program()
+        break
```

Comment:
please review the assignment instructions - you should be using a switch dict here